### PR TITLE
feat: Use env var for GQL rate limit instead of hard code val

### PR DESCRIPTION
**Pull request by ajay-sentry**
### Purpose/Motivation

This PR aims to move the hard coded gql rate limit value to an env variable that can be set per environment

In the same PR I've updated the staging limit to be 1000 from 300

### Notes to Reviewer
Anything to note to the team? Any tips on how to review, or where to start?

### Legal Boilerplate

Look, I get it. The entity doing business as "Sentry" was incorporated in the State of Delaware in 2015 as Functional Software, Inc. In 2022 this entity acquired Codecov and as result Sentry is going to need some rights from me in order to utilize my contributions in this PR. So here's the deal: I retain all rights, title and interest in and to my contributions, and by keeping this boilerplate intact I confirm that Sentry can use, modify, copy, and redistribute my contributions, under Sentry's choice of terms.

### CODE REVIEW

File: graphql_api/views.py [hunk 1]
```diff
@@ -320,7 +320,7 @@ def _check_ratelimit(self, request):
             user_ip = self.get_client_ip(request)
             key = f"rl-ip:{user_ip}"
 
-        limit = 300
+        limit = settings.GRAPHQL_RATE_LIMIT_VALUE
         window = 60  # in seconds
```
Comment (suejung-sentry) [inline graphql_api/views.py:324]: May be nice to have an `GRAPHQL_RATE_LIMIT_ENABLED` knob as well if not too annoying to add in. May be easier for DEC/self-hosted to use than trying to guess a large number for their use case :shrug:

File: codecov/settings_base.py [hunk 2]
```diff
@@ -92,13 +92,20 @@
...
-# Database
-# https://docs.djangoproject.com/en/2.1/ref/settings/#databases
+
+# GraphQL
 
 GRAPHQL_QUERY_COST_THRESHOLD = get_config(
     "setup", "graphql", "query_cost_threshold", default=10000
 )
 
+GRAPHQL_RATE_LIMIT_VALUE = get_config(
```
Comment (suejung-sentry) [inline codecov/settings_base.py:102]: Thoughts on a name that includes the units? Like GRAPHQL_RATE_LIMIT_RPM or similar?

**Review by suejung-sentry — APPROVED** [review]
approved! left optional comments

File: codecov/settings_base.py (hunk 2 above)
Comment (ajay-sentry) [reply codecov/settings_base.py:102]: Oooo I like that, I was straining my brain thinking of a name besides "Value" lol

File: graphql_api/views.py (hunk 1 above)
Comment (ajay-sentry) [reply graphql_api/views.py:324]: Fair point! I like it

**Review by suejung-sentry — APPROVED** [review]
nice!!

**Pull request merged**